sw/host/tests/crypto/ecdsa_kat/src/main.rs

```diff
@@ -240,12 +310,147 @@ fn run_ecdsa_testcase(
     }
     .send(&*uart)?;
 
-    let ecdsa_output = CryptotestEcdsaVerifyOutput::recv(&*uart, opts.timeout, false)?;
-    let success = match ecdsa_output {
-        CryptotestEcdsaVerifyOutput::Success => true,
-        CryptotestEcdsaVerifyOutput::Failure => false,
-        CryptotestEcdsaVerifyOutput::IntValue(i) => panic!("Invalid ECDSA verify result: {}", i),
-    };
+    CryptotestEcdsaPrivateKey {
+        d0: ArrayVec::try_from(d0.as_slice()).unwrap(),
+        d0_len: d0.len(),
+        d1: ArrayVec::try_from(d1.as_slice()).unwrap(),
+        d1_len: d1.len(),
+        unmasked_len: d0.len(),
+    }
+    .send(&*uart)?;
+    let success =
+        match operation {
+            CryptotestEcdsaOperation::Sign => {
+                let mut output_signature =
+                    CryptotestEcdsaSignature::recv(&*uart, opts.timeout, false)?;
+                // Truncate signature values to correct size for curve and convert to big-endian
+                output_signature.r.truncate(output_signature.r_len);
+                output_signature.s.truncate(output_signature.s_len);
+                output_signature.r.reverse();
+                output_signature.s.reverse();
+                match test_case.curve.as_str() {
+                    "p256" => {
+                        // Convert qx and qy to big-endian (resize in case to full length beforehand
+                        // in case there are leading zero(es))
+                        qx.resize(ECDSA_CMD_MAX_COORDINATE_BYTES_P256, 0u8);
+                        qy.resize(ECDSA_CMD_MAX_COORDINATE_BYTES_P256, 0u8);
+                        qx.reverse();
+                        qy.reverse();
+                        // Zero-fill the remaining space for the public key parameters, because
+                        // p256 requires an exact size
+                        // Verify the signature with the public key
+                        p256::ecdsa::VerifyingKey::from_encoded_point(
+                        &p256::EncodedPoint::from_affine_coordinates(
```

Comment:
Is there in an indentation problem here? The call to `from_affine_coordinates(` should should be shifted right by one right? I think this continues all the way down to the end of the `p256` match arm.

Reply:
This is fixed by moving the sign verification code to helper functions, per your below comment. Must have hit some strange behavior in `rustfmt.`